### PR TITLE
Update `Janitor`

### PR DESCRIPTION
**Pull request by christopher-buss**
Update to 1.15.4.

### CODE REVIEW

**Comment by OverHash** [comment]
Let's update this PR to the upstream https://github.com/howmanysmall/Janitor/releases/tag/v1.15.4 version -- rather than just the part of the -- just to be in-line with the releases of the official Janitor.

**Comment by christopher-buss** [comment]
I'm happy with this - it wasn't done as a realease when I made the PR

File: janitor/src/init.lua [hunk 1]
```diff
@@ -806,6 +794,7 @@ end
 export type Janitor = {
 	ClassName: "Janitor",
 	CurrentlyCleaning: boolean,
+	SuppressInstanceReDestroy: boolean,
```
Comment (OverHash) [inline janitor/src/init.lua:797]: We need to have this type somewhere in our Janitor class so that people can modify it if they want.

**Review by OverHash — CHANGES_REQUESTED** [review]
small things!

**Review by OverHash — APPROVED** [review]

**Pull request merged**